sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Custom/FastCreate.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers
+{
+    public partial class PostgreSqlFlexibleServerCollection
+    {
+        private GetCachedServerNameRestOperations _getCachedServerNameRestClient;
+
+        private GetCachedServerNameRestOperations GetCachedServerNameRestClient => _getCachedServerNameRestClient ??= new GetCachedServerNameRestOperations(Pipeline, Diagnostics.ApplicationId, Endpoint);
+
+        /// <summary>
+        /// Creates a new server using fast provisioning method.
+        /// </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="data"> The required parameters for creating a server. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<ArmOperation<PostgreSqlFlexibleServerResource>> FastCreateAsync(WaitUntil waitUntil, PostgreSqlFlexibleServerData data, CancellationToken cancellationToken = default)
```

Comment:
Could you please explain which part in the original generated code we are trying to fix using this piece of customized code?
If there is anything we could fix it using configuration, we should use configuration instead of using customized code because customized code will be hard to maintain without proper comments to explain why they are here

Reply:
The purpose of `FastCreate` custom function is to get a server name using `GetCachedServerName` and then use it for server `Create`, this server name allows for a fast creation. I'm not sure if this can be customized through configuration, any thoughts?

cc @rajsell @DaeunYim

Reply:
Let me ask in this way - do we have this method in the generated code if we do not have this piece of customized code?
And if we do, what is the difference between the generated method and this method?
I assume if we have a RestOperation class for your operation `GetCachedServerNameRestOperations` the generator should be able to generate that method, and your customization should be trying to fix something in that generated method

Reply:
No, the method `FastCreate` doesn't exist in the generated code. This new method contains a new logic that is a combination of both `GetCachedServerName` and `Create`.